File: apimodel/src/main/java/com/pgssoft/mvvm/model/api/ApiRate.java
package com.pgssoft.mvvm.model.api;

import com.google.gson.annotations.SerializedName;

public class ApiRate {
    @SerializedName("code")
    private String currencyCode;

    @SerializedName("currency")
    private String currencyName;

    @SerializedName("mid")
    private double rate;

    public ApiRate(String currencyCode, String currencyName, double rate) {
        this.currencyCode = currencyCode;
        this.currencyName = currencyName;
        this.rate = rate;
    }

    public String getCurrencyCode() {
        return currencyCode;
    }

    public String getCurrencyName() {
        return currencyName;
    }

    public double getRate() {
        return rate;
    }
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/model/api/ApiRate.kt
package com.pgssoft.kotlinplayground.model.api

import com.google.gson.annotations.SerializedName

class ApiRate(@SerializedName("code") val currencyCode: String,
              @SerializedName("currency") val currencyName: String,
              @SerializedName("mid") val rate: Double)
